### DISCUSSION TITLE
Utilizing the creative minds of the FOSS community with an ongoing bounty framework

### DISCUSSION THREAD

**Discussion by LPX55** [shapeshift/web#1080] [Ideas]
1. There are developers constantly on the lookout for new bounties, waiting for the chance to pounce on one when posted
2. There are developers, frontend devs, UX/UI designers, and biz devs with great ideas on improving the platform

Perhaps there should be a framework for these aforementioned groups which allows self-initiated bounties through a proposal for improvements/features. Obviously the proposed change must be approved by the workstream, and paid out only upon a successful merge into production. This could potentially accelerate platform development by multitudes. An example of something similar to this would be Arweave and their Open Web Foundry programs which has proved to be highly successful in the adoption and expansion of their ecosystem. 

Just a thought, feel free to chime in.

**Comment by 0xApotheosis** [comment]
I like this idea - we are capped more on the supply side than the demand side.

We could consider encouraging anyone with ideas to create an issue using a "Contributor-made bounty" template that we create, which requires a `bounty-approved` tag from the relevant workstream before it is posted to GitCoin.

The worksteams would need to stay vigilant on ensuring that the quality of each bounty description meets the standard before approving - the `bounty-approved` tag should have a high bar, and we'd need to explicitly define what that bar is (all relevant context provided, clear acceptance criteria, solves an actual issue, suitable for bounty hunters etc).

I suspect @0xdef1cafe would appreciate initiatives such as this that result in more bounties created.

**Comment by 0xdef1cafe** [comment]
@LPX55 this is a great idea.

shortly the engineering workstream as a whole will be responsible for the creation and posting of bounties, rather than a select few individuals, as we've refined the process and felt where the pain points are, so the supply side will increase on that front.

we've had some good PR comments and suggestions from previous bounty hunters, and need to foster this more.

bootstrapping the demand side is a harder problem - devs with the willingness and ability to contribute typically find us via discord, but we could definitely use some copy around being open to the idea of bounties from a product and engineering perspective.

this is really more of a concern of the product team at first, as most bounties will be "i'm a dev and would like x feature", we've yet to have devs say "i worked on y and we should consider doing z", but this will happen over time.

**Comment by cjthompson** [comment]
I would argue that there's nothing stopping anyone from doing this right now. The real roadblock is there isn't any "official" or documented process so people may not see that as an option.

I'd like to see a process added (into our `docs` repo) that someone can follow for submitting a proposal.

**Comment by willyogo** [comment]
boom, check this shiz out. was just showing this to the product stream yesterday, I think this could be the perfect tool to enable what you're describing @LPX55. Check out 1hive's demo for Gardens and lmk what y'all think.

reposting this post from Paul in Giveth's discord. Looks like a pretty dope tool for feature requests 👀 

"Hello Giveth!

1Hive Gardens is now using DeWork - a web3 project management tool that I think you guys will like. 

The relevant tool for the Giveth community is “Community Suggestions,” where anyone can request features and upvote other suggestions that they like. This will help the Gardens team get a sense of community interest in these features so they can be prioritized.

The Gardens team can also be funding some suggestions as bounties, so for anyone interested in contributing to Gardens that’s a good way to get started.

Here’s the invite link to join the DeWork project:  https://app.dework.xyz/o/1hive-4buSPaVwe8zjEjGnSd1it1/p/gardens-1ZW1QITfSxw1EJmkTMt797?inviteId=9241e39f-c069-4288-98dc-95250f92b28a

Please join and upvote any suggestions you really want! 🌱"

**Reply by LPX55** [reply]
@willyogo I've seen dework before and thought it was very neat! Would be super cool to see ShapeShift utilize it and see what it does to dev contributions
